app/api/controllers/validators/resolvers/ResolveEndDate.scala

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package api.controllers.validators.resolvers
+
+import api.models.errors.{EndDateFormatError, MtdError}
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+
+import java.time.LocalDate
+
+object ResolveEndDate extends Resolver[String, LocalDate] {
+  private val maxYear: Int = 2100
```

Comment:
ResolveEndDate could be turned into a class and take a maxYear arg, would make these more reusable.